### PR TITLE
Add SHA256 hash function

### PR DESCRIPTION
**Pull request by PlaceReporter99**
I'm surprised Vyxal does not have this yet!

### CODE REVIEW

**Review by nayakrujul — APPROVED** [review]
Cool. Looks good.

File: vyxal/elements.py [hunk 1]
```diff
@@ -449,6 +450,7 @@ def process_element(
         "lhs = pop(stack, 1, ctx); vy_type(lhs) == NUMBER_TYPE and time.sleep(lhs)",
         1,
     ),
+    "ø%": process_element("sha256(lhs.encode()).hexdigest()", 1),
```
Comment (ysthakur) [inline vyxal/elements.py:453]: I'd prefer if this were implemented as a function with the `@element` decorator, something like
```python
@element("ø%", 1)
def hash(lhs, ctx):
    """Element ø%
    ...
    """
    if isinstance(lhs, str):
        return sha256(lhs.encode()).hexdigest()
    else:
        <raise some kind of error saying lhs must be a string>
```

**Comment by ysthakur** [comment]
This is a good PR, it's nice that you added a test. What's the use case for it, though? Do you have any answers where SHA-256 would've been useful?

File: vyxal/elements.py [hunk 2]
```diff
@@ -5858,6 +5859,14 @@ def separate_runl_encode(lhs, ctx: Context):
     return lengths
 
 
+@element("ø%", 1)
+def sha256_hash(lhs, ctx):
+    """Element ø%
+    ...
```
Comment (ysthakur) [inline vyxal/elements.py:5865]: Make sure to put an actual element description here

File: vyxal/elements.py [hunk 3]
```diff
@@ -5858,6 +5859,14 @@ def separate_runl_encode(lhs, ctx: Context):
...
+    """
+    return sha256(str(lhs).encode()).hexdigest()
```
Comment (ysthakur) [inline vyxal/elements.py:5867]: A check for whether `lhs` is a string or not would be helpful, so users don't get weird errors saying `encode` can't be called on a `list` or whatever

**Review by ysthakur — CHANGES_REQUESTED** [review]

File: vyxal/elements.py (hunk 2 above)
Comment (nayakrujul) [reply vyxal/elements.py]: Yeah, just realised

File: vyxal/elements.py (hunk 3 above)
Comment (nayakrujul) [reply vyxal/elements.py:5867]: It converts to `str` anyway:

> `str(lhs).encode()`

File: vyxal/elements.py [hunk 4]
```diff
@@ -5858,6 +5859,14 @@ def separate_runl_encode(lhs, ctx: Context):
     return lengths
 
 
+@element("ø%", 1)
+def sha256_hash(lhs, ctx):
+    """Element ø%
+    (any) -> hash a using SHA256
```
Comment (ysthakur) [inline vyxal/elements.py:5865]:
```suggestion
    (str) -> hash a string using SHA256
```
This only works on strings, not any value (I guess you could do `str(lhs).encode()`, but I don't see much point in bothering)

File: vyxal/elements.py (hunk 3 above)
Comment (ysthakur) [reply vyxal/elements.py:5867]: Ah, my bad, didn't notice you did `str(lhs)`

File: vyxal/elements.py (hunk 4 above)
Comment (nayakrujul) [reply vyxal/elements.py:5865]: It *does* convert to string. `str(lhs).encode()` is in the code already.

Comment (ysthakur) [inline vyxal/elements.py:5865]:
```suggestion
    (any) -> hash a value using SHA256 (after converting to string)
```

**Review by ysthakur — APPROVED** [review]
I'm not sure when this would be helpful but doesn't hurt to have it. Good work PlaceReporter99 and nayakrujul!

**Review by Lyxal — APPROVED** [review]
Very good

**Pull request merged**